hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -347,9 +350,11 @@ private boolean executeHttpOperation(final int retryCount,
 
     LOG.debug("HttpRequest: {}: {}", operationType, httpOperation);
 
-    if (client.getRetryPolicy().shouldRetry(retryCount, httpOperation.getStatusCode())) {
-      int status = httpOperation.getStatusCode();
-      failureReason = RetryReason.getAbbreviation(null, status, httpOperation.getStorageErrorMessage());
+    int status = httpOperation.getStatusCode();
+    failureReason = RetryReason.getAbbreviation(null, status, httpOperation.getStorageErrorMessage());
+    retryPolicy = client.getRetryPolicy(failureReason);
+
+    if (retryPolicy.shouldRetry(retryCount, httpOperation.getStatusCode())) {
```

Comment:
This is overlapping with Anmol's question later .. the call that happens around line 344
` intercept.updateMetrics(operationType, httpOperation);`
will consider socket exceptions as throttling. When read timeout happens at the moment, we know of cases there it might be better to consider so. 
But if failure is due to connection timeout, then we dont want ThrottlingInterceptor to consider this as an input to increment throttling related metrics.

Reply:
Taken